src/kakarot/instructions/system_operations.cairo

```diff
@@ -467,10 +538,74 @@ namespace CreateHelper {
             bytecode=ctx.return_data,
         );
         local ctx: model.ExecutionContext* = ExecutionContext.update_sub_context(ctx.calling_context, ctx);
+        
+        // Append contracts to selfdestruct to the calling_context
+        Helpers.fill_array(
+            fill_len=ctx.sub_context.destroy_contracts_len,
+            input_arr=ctx.sub_context.destroy_contracts,
+            output_arr=ctx.destroy_contracts + ctx.destroy_contracts_len,
+        );
+        let ctx = ExecutionContext.update_destroy_contracts(
+            self=ctx,
+            destroy_contracts_len=ctx.destroy_contracts_len + ctx.sub_context.destroy_contracts_len,
+            destroy_contracts=ctx.destroy_contracts,
+            stop=ctx.stopped,
+        );
+
         let (address_high, address_low) = split_felt(ctx.sub_context.evm_contract_address);
         let stack = Stack.push(ctx.stack, Uint256(low=address_low, high=address_high));
         let ctx = ExecutionContext.update_stack(ctx, stack);
 
         return ctx;
     }
 }
+
+namespace SelfDestructHelper {
+
+    // @notice helper for SELFDESTRUCT operation.
+    // @notice overwrite contract account bytecode with 0s
+    // @notice remove contract from registry
+    // @return The pointer to the updated execution_context.
+    func selfdestruct{
```

Comment:
could we name this this `finalize`? to be similar to the other helpers, and because the `selfdestruct` is already the name of the helper?